### PR TITLE
String strip! family

### PR DESCRIPTION
**Pull request by RobertBendun**
Another 3 for #217

### CODE REVIEW

File: src/string_object.cpp [hunk 1]
```diff
@@ -671,6 +679,32 @@ Value StringObject::lstrip(Env *env) const {
...
+    for (first_char = 0; first_char < len; first_char++) {
+        char c = c_str()[first_char];
+        if (!is_strippable_whitespace(c))
+            break;
+    }
+
+    if (first_char == 0)
+        return NilObject::the();
+
+    memcpy(&m_string[0], &m_string[0] + first_char, len - first_char);
```
Comment (seven1m) [inline src/string_object.cpp:703]: You cannot memcpy from one part of a string onto an overlapping part. From the man page for `memcpy`:

>If dst and src overlap, behavior is undefined.

So we'll need a temporary buffer.

**Review by seven1m — CHANGES_REQUESTED** [review]

File: src/string_object.cpp (hunk 1 above)
Comment (RobertBendun) [reply src/string_object.cpp:703]: Oh yes, wierldy I have been sure that is defined for this kind of copy, but we can use [memmove](https://en.cppreference.com/w/cpp/string/byte/memmove) without temporary buffer

Comment (seven1m) [reply src/string_object.cpp]: OK, I think I see. It's not actually overlapping, since it's copying characters to the "left". Ok, carry on.

**Review by seven1m — APPROVED** [review]

**Comment by seven1m** [comment]
Thanks!

**Pull request merged**